### PR TITLE
Add soft keywords

### PR DESCRIPTION
**Pull request by gvanrossum**
These are like keywords but they only work in context; they are not reserved except when there is an exact match.

This would enable things like match statements without reserving `match` (which would be bad for the `re.match()` function and probably lots of other places).

Automerge-Triggered-By: @gvanrossum

### CODE REVIEW

**Comment by gvanrossum** [comment]
Looks like the failing tests all have to do with print statements.
```
    test_exceptions test_future test_grammar test_print
    test_source_encoding
```

**Comment by gvanrossum** [comment]
Fixed an unfortunate interaction with the new `D()` macro calls. Deleted the print statement easter egg. Now ready for review. (How to test this other than by adding a print statement?)

**Comment by pablogsal** [comment]
> How to test this other than by adding a print statement?

A weak test would be adding something like [this](https://github.com/python/cpython/blob/master/Lib/test/test_peg_generator/test_c_parser.py#L379-L392) that uses the double quotes and checks that `_PyPegen_expect_soft_keyword` is generated in the source and that something with single quotes  does not include `_PyPegen_expect_soft_keyword` in the source.

**Comment by gvanrossum** [comment]
Now with tests. I'd like to keep this as an easter egg, hence the skip issue label.

File: Lib/test/test_peg_generator/test_c_parser.py [hunk 1]
```diff
@@ -402,3 +402,34 @@ def test_error_in_rules(self) -> None:
...
+        start: expr+ NEWLINE? ENDMARKER
+        expr: "foo"
+        """
+        grammar = parse_string(grammar_source, GrammarParser)
+        parser_source = generate_c_parser_source(grammar)
+        assert "expect_soft_keyword" in parser_source
+
+    def test_soft_keywords_parse(self) -> None:
+        grammar_source = """
+        @subheader "#undef D\\n#define D(x) x\\n"
```
Comment (pablogsal) [inline Lib/test/test_peg_generator/test_c_parser.py]: Question: Why Is this needed? In the latest version this macro expands to nothing outside the interpreter core:

https://github.com/python/cpython/blob/master/Tools/peg_generator/pegen/c_generator.py#L32

**Review by pablogsal — APPROVED** [review]
LGTM!

I left one question

File: Lib/test/test_peg_generator/test_c_parser.py (hunk 1 above)
Comment (gvanrossum) [reply Lib/test/test_peg_generator/test_c_parser.py]: Whoops, that was a failed attempt at debugging. Deleting.
